source/common/quic/envoy_quic_utils.h

```diff
@@ -73,18 +73,21 @@ class HeaderValidator {
 template <class T>
 std::unique_ptr<T>
 quicHeadersToEnvoyHeaders(const quic::QuicHeaderList& header_list, HeaderValidator& validator,
-                          uint32_t max_headers_allowed, absl::string_view& details) {
+                          uint32_t max_headers_allowed, absl::string_view& details,
+                          quic::QuicRstStreamErrorCode& rst) {
```

Comment:
Instead of passing in one more argument, can we tell which reset error to use based on the output argument `details`?  If the return value is null, details_ must have been populated already by these two utility functions.